### ISSUE TITLE
X509Certificate with no password throws crypto exception when IIS pool does not have LoadUserProfile enabled

### ISSUE THREAD

**Issue by dk-mushiyoke** [tomasmcguinness/dotnet-passbook#110]
Pre-condition:
* Have a web service running in an IIS instance with LoadUserProfile disabled (default behavior) 
* Have a signing certificate with no import password

Steps to recreate:
* Attempt to create a pass by calling new PassGenerator().Generate(request)

Expected behavior
* Pass is created successfully

Actual behavior
* X509Certificate ctor throws an exception
* Message: The system cannot find the file specified
* Stacktrace:    at Internal.Cryptography.Pal.CertificatePal.FilterPFXStore(Byte[] rawData, SafePasswordHandle password, PfxCertStoreFlags pfxCertStoreFlags)\r\n   at Internal.Cryptography.Pal.CertificatePal.FromBlobOrFile(Byte[] rawData, String fileName, SafePasswordHandle password, X509KeyStorageFlags keyStorageFlags)\r\n   at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(Byte[] data)\r\n   at Passbook.Generator.PassGenerator.GetCertificateFromBytes(Byte[] bytes, String password)\r\n   at Passbook.Generator.PassGenerator.GetCertificate(PassGeneratorRequest request)\r\n   at Passbook.Generator.PassGenerator.ValidateCertificates(PassGeneratorRequest request)\r\n   at Passbook.Generator.PassGenerator.CreatePackage(PassGeneratorRequest request)\r\n   at Passbook.Generator.PassGenerator.Generate(PassGeneratorRequest request)\r\n

**Comment by tomasmcguinness** [comment]
I've merged your PR. Please let me know if this resolves your problem? I always thought the LoadUserProfile permission was required for working with certificates. I'll be happy if this resolves the issue!

**Comment by dk-mushiyoke** [comment]
Thanks @tomasmcguinness. I've already tested this change on my own branch and confirmed to be working. We switched from bundling the certificate to using an Azure Key Vault service, which decrypts the certificate at import time and always returns the raw certificate, which introduced this issue since the X509Certificate constructor were not setting those flags. The MachineKeySet flag lets you bypass checking user profiles on an IIS instance which I had very little control over.
